bin/init.js

```diff
@@ -1,188 +1,317 @@
-const fgCyan = '\x1b[36m';
-
-// Functions
-const consoleOutput = ( color, text ) => {
-	console.log( color, text );
+#! /usr/bin/env node
+
+/* eslint no-console: 0 */
+
+/**
+ * External dependencies
+ */
+const fs = require('fs');
+const path = require('path');
+const readline = require('readline');
+
+/**
+ * Define Constants
+ */
+const rl = readline.createInterface({
+	input: process.stdin,
+	output: process.stdout,
+});
+const info = {
+	error: (message) => {
+		return `\x1b[31m${ message }\x1b[0m`;
+	},
+	success: (message) => {
+		return `\x1b[32m${ message }\x1b[0m`;
+	},
+	warning: (message) => {
+		return `\x1b[33m${ message }\x1b[0m`;
+	},
+	message: (message) => {
+		return `\x1b[34m${ message }\x1b[0m`;
+	},
 };
+let fileContentUpdated = false;
+let fileNameUpdated = false;
+
+// Start with a prompt.
+rl.question('Would you like to setup the theme? (Y/n) ', (answer) => {
+	if ('n' === answer.toLowerCase()) {
+		console.log(info.warning('\nTheme Setup Cancelled.\n'));
+		process.exit(0);
+	}
+	rl.question('Enter theme name (shown in WordPress admin)*: ', (themeName) => {
+		const themeInfo = setupTheme(themeName);
+		rl.question('Confirm the Theme Details (Y/n) ', (confirm) => {
+			if ('n' === confirm.toLowerCase()) {
+				console.log(info.warning('\nTheme Setup Cancelled.\n'));
+				process.exit(0);
+			}
+			initTheme(themeInfo);
+			rl.close();
+		});
+	});
+});
+
+rl.on('close', () => {
+	process.exit(0);
+});
+
+/**
+ * Theme Setup
+ *
+ * @param {string} themeName
+ *
+ * @return {Object} themeInfo
+ */
+const setupTheme = (themeName) => {
+	console.log(info.success('\nFiring up the theme setup...'));
+
+	// Ask theme name.
+	if (!themeName) {
+		console.log(info.error('\nTheme name is required.\n'));
+		process.exit(0);
+	}
+
+	// Generate theme info.
+	const themeInfo = generateThemeInfo(themeName);
 
-const findReplace = ( findString, replaceString ) => {
-	const regex = new RegExp( findString, 'g' );
-	const options = {
-		files: `${ rootDir }/**/*`,
-		from: regex,
-		to: replaceString,
-		ignore: [
-			`${ rootDir }/node_modules/**/*`,
-			`${ rootDir }/.git/**/*`,
-			`${ rootDir }/.github/**/*`,
-			`${ rootDir }/vendor/**/*`,
-			`${ rootDir }/bin/init.js`,
-		],
+	const themeDetails = {
+		'Theme Name: ': `${ themeInfo.themeName }`,
+		'Theme Version: ': `1.0.0`,
+		'Text Domain: ': `${ themeInfo.kebabCase }`,
+		'Package: ': `${ themeInfo.trainCase }`,
+		'Namespace: ': `${ themeInfo.pascalSnakeCase }`,
+		'Function Prefix: ': `${ themeInfo.snakeCaseWithUnderscoreSuffix }`,
+		'CSS Class Prefix: ': `${ themeInfo.kebabCaseWithHyphenSuffix }`,
+		'PHP Variable Prefix: ': `${ themeInfo.snakeCaseWithUnderscoreSuffix }`,
+		'Version Constant: ': `${ themeInfo.macroCase }_VERSION`,
+		'Theme Directory Constant: ': `${ themeInfo.macroCase }_TEMP_DIR`,
+		'Theme Build Directory Constant: ': `${ themeInfo.macroCase }_BUILD_DIR`,
+		'Theme Build Directory URI Constant: ': `${ themeInfo.macroCase }_BUILD_URI`,
```

Comment:
```suggestion
    const themeDetails = {
		'Theme Name: ': `${ themeInfo.themeName }`,
		'Theme Version: ': '1.0.0',
		'Text Domain: ': themeInfo.kebabCase,
		'Package: ': themeInfo.trainCase,
		'Namespace: ': themeInfo.pascalSnakeCase,
		'Function Prefix: ': themeInfo.snakeCaseWithUnderscoreSuffix,
		'CSS Class Prefix: ': themeInfo.kebabCaseWithHyphenSuffix,
		'PHP Variable Prefix: ': themeInfo.snakeCaseWithUnderscoreSuffix,
		'Version Constant: ': `${ themeInfo.macroCase }_VERSION`,
		'Theme Directory Constant: ': `${ themeInfo.macroCase }_TEMP_DIR`,
		'Theme Build Directory Constant: ': `${ themeInfo.macroCase }_BUILD_DIR`,
		'Theme Build Directory URI Constant: ': `${ themeInfo.macroCase }_BUILD_URI`,
	};
```
Don't use template literals where not required.

Reply:
Yes, there is no use of template literals here.